google/resource_storage_bucket_iam_test.go

```diff
@@ -144,7 +144,7 @@ data "google_iam_policy" "fooPolicy" {
 }
 
 resource "google_storage_bucket_iam_policy" "foo" {
-  bucket      = "${google_storage_bucket.%s}"
+  bucket      = "${google_storage_bucket.%s.name}"
```

Comment:
This still isn't quite right - you just want "${google_storage_bucket.bucket.name}", right?  You just want the name of that bucket?

Reply:
`%s` here is the name of the bucket (the value of `google_storage_bucket.bucket.name`), right?

Reply:
```
resource "google_storage_bucket" "bolcom-sbx-kroonapp-010-kroon" {
  name          = "bolcom-sbx-kroonapp-010-kroon"
  project       = "bolcom-sbx-kroonapp-010"
  location      = "EUROPE-WEST1"
  storage_class = "REGIONAL"
  force_destroy = false

  labels = {
    opex            = "teamprovisioning"
    stores_pii_data = "false"
  }

  versioning = {
    enabled = false
  }
}

data "google_iam_policy" "kroontest" {
  binding {
    role = "roles/storage.objectAdmin"

    members = [
      "user:rkroon@bol.com",
    ]
  }
}

resource "google_storage_bucket_iam_policy" "kroontest" {
  bucket      = "${google_storage_bucket.bolcom-sbx-kroonapp-010-kroon.name}"
  policy_data = "${data.google_iam_policy.kroontest.policy_data}"
}
```

This is the generated terraform file that works for me.

Reply:
Right, because the bucket resource is called "bolcom-sbx-kroonapp-010-kroon", but that's also its name attribute.  In the test, the bucket resource is called "bucket", but its name attribute is a random string.  You want to access that random string here.  I'm rerunning the tests now.